### PR TITLE
[Feature:SubminiPolls] Frontend poll timer

### PR DESCRIPTION
**Pull request by Kush-Raval**
### Please check if the PR fulfills these requirements:

* [x] Tests for the changes have been added/updated (if possible)
* [x] Documentation has been updated/added if relevant
* [x] Screenshots are attached to Github PR if visual/UI changes were made

### What is the current behavior?

The backend for the poll timer exists but there is no way to see the real-time time of the timer in both the instructor and the student pages. 
### What is the new behavior?

See the timer in the viewPoll page for both instructors and students.

image

image

### Other information?

Test when the poll ends if the timer is synced correctly.
The timer will be visible on the ViewPoll page. 
This feature works toward completing https://github.com/Submitty/Submitty/issues/5913.

### CODE REVIEW

**Review by JWei17 — APPROVED** [review]
The counter for poll timer display correct numbers, and it looks clearly on light mod and dark mod.  Approve

**Review by ryvaru — CHANGES_REQUESTED** [review]
image

Tested for functionality and the code works as intended for the most part- the poll correctly displays a notification that it has been closed after the allotted time has passed. 

However, there is a visual error where polls that did not enable timer show a notification that "Timer has expired" when no timer was actually set, this should be fixed.

File: site/app/templates/polls/ViewPoll.twig
Comment (powe97) [inline site/app/templates/polls/ViewPoll.twig:111]: You should use Twig to do this with `{% if timer_enabled}`.

**Review by powe97 — CHANGES_REQUESTED** [review]

**Review by powe97 — COMMENTED** [review]
Looks good, but I would remove the indent from the code inside the `{% if %}` block to be consistent with code further down.

**Review by powe97 — APPROVED** [review]
Looks good

**Comment by Kush-Raval** [comment]
The PR is at its finishing touches, I think it would be a good idea to add cypress tests in the PR since polls is well-maintained.

It would be simple checks such as: if timer isn't enabled it shouldn't display anything, if timer hits 00:00 it displays the correct message "poll is ended" etc.

**Review by micpap25 — APPROVED** [review]
Tests pass and look good to me

File: site/app/templates/polls/ViewPoll.twig
Comment (ziesski) [inline site/app/templates/polls/ViewPoll.twig:128]: Using `innerHTML` poses a security risk because user input can be parsed as HTML, leading to potential HTML injection and harmful script execution. Use `textContent` or `text()` instead of `html()` or `innerHTML` to prevent this.

File: site/app/templates/polls/ViewPoll.twig
Comment (ziesski) [inline site/app/templates/polls/ViewPoll.twig:110]: Could we move this to js file so that it can get linted? if js file does not exist lets make one.

File: site/app/templates/polls/ViewPoll.twig
Comment (ziesski) [inline site/app/templates/polls/ViewPoll.twig:148]: same here

**Review by ziesski — DISMISSED** [review]

File: site/app/templates/polls/ViewPoll.twig
Comment (micpap25) [reply site/app/templates/polls/ViewPoll.twig:128]: Does this apply if the `innerHTML` is not taking user input?

Comment (ziesski) [reply site/app/templates/polls/ViewPoll.twig:128]: Yes, it is good practice to avoid innerHTML whenever possible, even when content is not user-generated. Better future proof, and consistency across the codebase.

File: site/app/templates/polls/ViewPoll.twig
Comment (Kush-Raval) [reply site/app/templates/polls/ViewPoll.twig:124]: or .css()

**Review by Kush-Raval — COMMENTED** [review]
Almost finished!
For me, when I set the time to a small amount like 30 seconds there is considerable amount of clock skew. This might be my virtual machine that is at issue. I think there is VM clock skew but I'm not sure if that is the problem.

File: site/app/templates/polls/ViewPoll.twig
Comment (zacharymnp) [reply site/app/templates/polls/ViewPoll.twig]: done in [70785f6](https://github.com/Submitty/Submitty/pull/10337/commits/70785f627ffdd44e5beed80aa95e15de48a82be3)

File: site/app/templates/polls/ViewPoll.twig
Comment (zacharymnp) [reply site/app/templates/polls/ViewPoll.twig]: done in [9d0a292](https://github.com/Submitty/Submitty/pull/10337/commits/9d0a292e16e2fac65097cd47d3baf36fc2b03c58)

File: site/public/css/polls.css
Comment (Kush-Raval) [inline site/public/css/polls.css:42]: I think with your new changes we don't use the .header anymore

**Review by Kush-Raval — COMMENTED** [review]
Works great I would just change the small fix, I think it is ready for a maintainer to look at.
Functions as intended for both students and instructors.

File: site/public/css/polls.css
Comment (zacharymnp) [reply site/public/css/polls.css]: done in [a6df8cc](https://github.com/Submitty/Submitty/pull/10337/commits/a6df8ccacae3e0075b3a3e2eca960331c4b7f316)

**Comment by zacharymnp** [comment]
As discussed in meeting, we will leave the remaining functions in the Twig files and add an issue to improve it later.

File: site/app/templates/polls/ViewPoll.twig
Comment (IDzyre) [inline site/app/templates/polls/ViewPoll.twig:10]:
```suggestion
        <h2 id="timer" class="timer" style="display:none" data-testid="timer">
```

File: site/cypress/e2e/Cypress-Feature/polls.spec.js
Comment (IDzyre) [inline site/cypress/e2e/Cypress-Feature/polls.spec.js:122]:
```suggestion
        cy.get('[data-testid="timer"]').contains('Poll Ended');
```
Same with other timers

**Review by IDzyre — CHANGES_REQUESTED** [review]

**Review by williamjallen — APPROVED** [review]
Code looks good to me.  I tested manually and did not observe any issues.  In the future, we could relocate the timer text to somewhere more intuitive, but the current design is good enough for now.

Great work on this @Kush-Raval, and thanks for taking this over @zacharymnp!

A screenshot from my successful test:
image

**Comment by williamjallen** [comment]
@zacharymnp It looks like a selector needs to be updated in the polls test.  This should be good to go once that's done.

**Review by ziesski — APPROVED** [review]
Reviewed, looks good

**Pull request merged**
